File: tests/src/helpers/responses.ts
import { Organization } from "./types";

namespace RS {
  export namespace auth {
    export interface signin {
      accessToken: string;
      refreshToken: string;
    }
    export interface refresh {
      accessToken: string;
      refreshToken: string;
    }
  }
  export namespace users {
    export interface current {
      id: number;
      firstName: string;
      email: string;
    }
    export interface get_organizations {
      organizations: Organization[];
    }
  }
  export namespace organizations {
    export interface get extends Organization {};
    export interface get_members {
      members: number[];
    };
  }
  export namespace test {
    export interface signup {
      accessToken: string;
      refreshToken: string;
    }
    export namespace organizations {
      export interface create extends Organization {}
      export interface invite {}
    }
  }
}

export default RS;

File: src/typings/responses.py
# pylint: disable=invalid-name
from __future__ import annotations

from utils.schemas import BaseModel
from typings import types


class RS:
    class auth:
        class signin(BaseModel):
            access_token: str
            refresh_token: str

        class refresh(BaseModel):
            access_token: str
            refresh_token: str

    class users:
        class current(BaseModel):
            id: int
            first_name: str
            email: str

        class get_organizations(BaseModel):
            organizations: list[types.Organization]

    class organizations:
        class get(types.Organization): ...

        class get_members(BaseModel):
            members: list[int]

    class test:
        class signup(BaseModel):
            access_token: str
            refresh_token: str

        class organizations:
            class create(types.Organization): ...